internal/agent/device/osimage.go

```diff
@@ -58,24 +59,24 @@ func (c *OSImageController) ensureImage(ctx context.Context, desired *v1alpha1.R
 		return nil
 	}
 
-	host, err := c.bootc.Status(ctx)
+	// TODO: handle the case where the host is reconciled but also in a dirty state (staged).
+	desiredImage := image.SpecToImage(desired.Os)
+	reconciled, err := c.isOsImageReconciled(ctx, desiredImage)
 	if err != nil {
 		return err
 	}
-
-	// TODO: handle the case where the host is reconciled but also in a dirty state (staged).
-	if container.IsOsImageReconciled(host, desired) {
+	if reconciled {
 		c.log.Debugf("Host is reconciled to os image %s", desired.Os.Image)
 		return nil
 	}
 
-	image := desired.Os.Image
-	c.log.Infof("Switching to os image: %s", image)
-	if err := c.bootc.Switch(ctx, image); err != nil {
+	c.log.Infof("Switching to os image: %s", desired.Os.Image)
+	target := desiredImage.ToBootcTarget()
```

Comment:
My understanding is we are converting the spec to image then back to something bootc can consume. But it really has what it needed before :). I could be missing something.